test/models/user_test.rb

```diff
@@ -43,40 +40,24 @@ def setup
     assert @user.invalid?
   end
 
+  test "user is not valid without address" do
+    @user.address = nil
+    assert @user.invalid?
+  end
+
   test "user is not valid without email" do
     @user.email = nil
     assert @user.invalid?
   end
 
   test "user has an optional username that is between 2 and 32 characters" do
-    @user1 = User.create(username: 'user',
-            password: 'password',
-            first_name: 'John',
-            last_name: 'Doe',
-            email: 'examples@example.com')
-
-    @user2 = User.create(username: 'us',
-            password: 'password',
-            first_name: 'John',
-            last_name: 'Doe',
-            email: 'example1@example.com')
-
-    @user3 = User.create(username: 'u',
-            password: 'password',
-            first_name: 'John',
-            last_name: 'Doe',
-            email: 'example5@example.com')
-
-    @user4 = User.create(username: 'ThisStringIs42CharactersLongBelieveItOrNot',
-            password: 'password',
-            first_name: 'John',
-            last_name: 'Doe',
-            email: 'example4@example.com')
+    user1 = build(:user)
+    user2 = build(:user, username: "u")
+    user3 = build(:user, username: 'ThisStringIs42CharactersLongBelieveItOrNot')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.